packages/material-ui/src/TextareaAutosize/TextareaAutosize.js

```diff
@@ -22,38 +22,41 @@ const styles = {
   },
 };
 
-/**
- * @ignore - internal component.
- *
- * To make public in v4+.
- */
-const Textarea = React.forwardRef(function Textarea(props, ref) {
+const TextareaAutosize = React.forwardRef(function TextareaAutosize(props, ref) {
   const { onChange, rows, rowsMax, style, value, ...other } = props;
 
   const { current: isControlled } = React.useRef(value != null);
   const inputRef = React.useRef(null);
-  const [state, setState] = React.useState({});
-  const shadowRef = React.useRef(null);
   const handleRef = useForkRef(ref, inputRef);
+  const shadowRef = React.useRef(null);
+  const [state, setState] = React.useState({});
 
   const syncHeight = React.useCallback(() => {
     const input = inputRef.current;
-    const inputShallow = shadowRef.current;
-
     const computedStyle = window.getComputedStyle(input);
+
+    const inputShallow = shadowRef.current;
     inputShallow.style.width = computedStyle.width;
     inputShallow.value = input.value || props.placeholder || 'x';
 
+    const boxSizing = computedStyle['box-sizing'];
+    const padding =
+      getStyleValue(computedStyle, 'padding-bottom') + getStyleValue(computedStyle, 'padding-top');
+    const border =
+      getStyleValue(computedStyle, 'border-bottom-width') +
+      getStyleValue(computedStyle, 'border-top-width');
+
     // The height of the inner content
     const innerHeight = inputShallow.scrollHeight;
-    const boxSizing = computedStyle['box-sizing'];
 
     // Measure height of a textarea with a single row
     inputShallow.value = 'x';
-    const singleRowHeight = inputShallow.scrollHeight;
+    let singleRowHeight = inputShallow.scrollHeight;
+    singleRowHeight -= padding;
 
     // The height of the outer content
-    let outerHeight = innerHeight;
+    let outerHeight = innerHeight - padding;
+    // console.log(1, { singleRowHeight, boxSizing, outerHeight });
```

Comment:
I take it you are removing the `console.log`s?

Reply:
Haha, it was here for me to debug. I love leaving console lot in my own code for futur usage. However, different context here. I'm removing them :+1: